File: python/p_035.py
import itertools
from sympy import isprime
from tqdm import tqdm

def get_circular_combinations(Number):
    digits = [int(d) for d in str(Number)]
    Combinations = []

    for i in range(len(digits)):
        rotated_digits = digits[i:] + digits[:i]

        if rotated_digits[0] != 0:
            Combinations.append(int("".join(map(str, rotated_digits))))
    
    return Combinations

def consist_even_digit(Number):
    digit_str = str(Number)
    for digit in str(Number):
        if int(digit) % 2 == 0 and int(digit) != 0:
            return True
    return False

NumberOfCircularPrimes = 4      # I won't check the number 2 in the for loop.
for N in tqdm(range(10, 1000000), desc="Processing numbers"):
    if consist_even_digit(N) == False:
        if isprime(N):
            Combinations = get_circular_combinations(N)
            k = 1
            for i in range(len(Combinations)):
                if isprime(Combinations[i]):
                    k *= 1
                else:
                    k *= 0
            if k == 1:
                print(N)
            NumberOfCircularPrimes += k
            

print(NumberOfCircularPrimes)

File: julia/p_035.jl
using Primes
using ProgressMeter

function get_circular_combinations(Number)
    digit_list = digits(Number)
    Combinations = []

    for i in 1:length(digit_list)
        rotated_digits = vcat(digit_list[i:end], digit_list[1:i-1])
        if rotated_digits[1] != 0
            push!(Combinations, parse(Int, join(rotated_digits)))
        end
    end
    return Combinations
end

function consist_even_digit(Number)
    digit_str = string(Number)
    for digit in digit_str
        if parse(Int, digit) % 2 == 0 && parse(Int, digit) != 0
            return true
        end
    end
    return false
end

NumberOfCircularPrimes = 4

@showprogress for N in 10:99999

    if !consist_even_digit(N)
        if isprime(N)
            Combinations = get_circular_combinations(N)
            k = 1
            for comb in Combinations
                if isprime(comb)
                    k *= 1
                else
                    k *= 0
                end
            end
            NumberOfCircularPrimes += k
        end
    end
end

println(NumberOfCircularPrimes)